edn/src/entities.rs

```diff
@@ -22,6 +22,19 @@ use symbols::{
     PlainSymbol,
 };
 
+use types::{
+    ValueAndSpan,
+};
+
+/// Entity and value places embed values, either directly (i.e., `ValuePlace::Atom`) or indirectly
+/// (i.e., `EntityPlace::LookupRef`).  In order to maintain the graph of `Into` and `From`
+/// relations, we need to ensure that `{Value,Entity}Place` can't match as a potential value.  This
```

Comment:
On first read, the dual meaning of "value" in this comment makes the web a little confusing at first read - but perhaps that's just my unfamiliarity with the subject. Unsure what you'd call "embedded values" other than values though.

Perhaps this can be talked about in terms of terminating value resolution? E.g. "{Entity,Value}Place need to resolve to a concrete value, which could happen either directly (i.e. ValuePlace::Atom, value is embedded in the enum) or indirectly (i.e. EntityPlace::LookupRef, value is a hop away). We need to ensure that the resolution is guaranteed to terminate, etc etc".